plugins/backend/backstage-plugin-argo-cd-backend/src/service/argocd.service.ts

```diff
@@ -116,16 +117,25 @@ export class ArgoService implements ArgoServiceApi {
   async findArgoApp(options: {
     name?: string;
     selector?: string;
+    namespace?: string;
   }): Promise<findArgoAppResp[]> {
     if (!options.name && !options.selector) {
       throw new Error('name or selector is required');
     }
     const resp = await Promise.all(
       this.instanceConfigs.map(async (argoInstance: any) => {
         let getArgoAppDataResp: any;
+        let token: string;
+        try {
+          token = argoInstance.token || (await this.getArgoToken(argoInstance));
+        } catch (error: any) {
+          this.logger.error(
+            `Error getting token from Argo Instance ${argoInstance.name}: ${error.message}`,
+          );
+          return null;
+        }
```

Comment:
I think try/catch for token acquisition/Application data acquisition should be separated in order to convey the error to the user in an understandable manner.